src/v/cluster/controller_backend.cc

```diff
@@ -853,6 +858,23 @@ ss::future<> controller_backend::add_to_shard_table(
       });
 }
 
+ss::future<bool> controller_backend::insert_to_shard_table(
+  model::ntp ntp, ss::shard_id shard, model::revision_id revision) {
+    return _shard_table.map_reduce0(
+      [self = _self, ntp = std::move(ntp), shard, revision](shard_table& s) {
+          vlog(
+            clusterlog.trace,
+            "[n: {}, r: {}] inserting {} to shard table at {}",
+            self,
+            revision,
+            ntp,
+            shard);
+          return s.insert(ntp, shard, revision);
+      },
+      false,
```

Comment:
Are you sure you want to know if any succeed, rather than if any fail?

I notice that the result is ignored, anyway: https://github.com/vectorizedio/redpanda/pull/2258/commits/648bf90107f3c860a6532e0ec4f5aafb38d28a2b#diff-33a20c72cf1f1964ca962b896b11a0831f7555ac5a528b91235c34e2686e9317R902

Reply:
Good point, will modify this to return true if all succeed

Reply:
Is there enough context to do anything about it? Would a `vector<bool>`, ok, maybe not a `vector<bool>`, but a set of results be preferable?

Reply:
Also more good points, the implications of failure would mean that the shard table for a particular shard doesn't have an entry for the materialized ntp that was attempted to be inserted. `controller_backend::add_to_shard_table` returns nothing, but thats because `shard_table::update` returns nothing as well.

Reply:
You [previously said you'd fixed a bug with `update`.](https://github.com/vectorizedio/redpanda/pull/2258#discussion_r706158618). I'm intrigued; what was the bug? What semantics do you require?

Reply:
> Also more good points, the implications of failure would mean that the shard table for a particular shard doesn't have an entry for the materialized ntp that was attempted to be inserted.

Doesn't failure of insert mean that the shard_table *does* already have an entry?

Reply:
Yes which really isn't a failure then... upon second look i agree this method should look just like `add_to_shard_table`.

To your other question, the bug with update was that I was passing in a raft id that was curried from the source topic, causing update to hit a `return` statement before elements were actually inserted into the shard table. This is why I am using the `insert` method.